openstates/ri/legislators.py

```diff
@@ -25,123 +31,72 @@ def scrape(self, chamber, term):
         if chamber == 'upper':
             url = ('http://webserver.rilin.state.ri.us/Documents/Senators.xls')
             rep_type = 'Senator'
-            source_url = 'http://www.rilin.state.ri.us/senators/default.aspx'
-            source_url_title_replacement = rep_type
             contact_url = 'http://webserver.rilin.state.ri.us/Email/SenEmailListDistrict.asp'
         elif chamber == 'lower':
             url = ('http://webserver.rilin.state.ri.us/Documents/Representatives.xls')
             rep_type = 'Representative'
-            source_url = 'http://www.rilin.state.ri.us/representatives/default.aspx'
-            source_url_title_replacement = 'Rep. '
             contact_url = 'http://webserver.rilin.state.ri.us/Email/RepEmailListDistrict.asp'
 
+        contact_page = self.lxmlize(contact_url)
+        contact_info_by_district = {}
+        for row in contact_page.xpath('//tr[@valign="TOP"]'):
+            tds = row.xpath('td')
+            (detail_link, ) = tds[link_col_ix].xpath('.//a/@href')
+            # Ignore name (2nd col). We have a regex built up below for the spreadsheet name I don't want to touch
+            district, _, email, phone = [td.text_content().strip() for td in tds[:link_col_ix]]
+            contact_info_by_district[district] = {
+                'email': email,
+                'phone': phone,
+                'detail_link': detail_link,
+            }
+
         self.urlretrieve(url, 'ri_leg.xls')
 
         wb = xlrd.open_workbook('ri_leg.xls')
         sh = wb.sheet_by_index(0)
 
-        # This isn't perfect but it's cheap and better than using the
-        # XLS doc as the source URL for all legislators.
-        # 374: RI: legislator url
-        leg_source_url_map = {}
-        leg_page = self.lxmlize(source_url)
-
-        for link in leg_page.xpath('//td[@class="ms-vb2"]'):
```

Comment:
The result of this snippet wasn't being used. This was modified into `contact_info_by_district`.